src/Symfony/Component/DependencyInjection/Compiler/FactoryReturnTypePass.php

```diff
@@ -65,6 +66,9 @@ private function updateDefinition(ContainerBuilder $container, $id, Definition $
         if (is_string($factory)) {
             try {
                 $m = new \ReflectionFunction($factory);
+                if (false !== $m->getFileName() && file_exists($m->getFileName())) {
+                    $container->addResource(new FileResource($m->getFileName()));
```

Comment:
yes, done in #21419 : https://github.com/symfony/symfony/pull/21419/files#diff-58acb002c7d222ac4537afd97530104bR85